gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/BaseGradleTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.squareup.anvil.plugin
+
+import com.rickbusarow.kase.gradle.GradleKotlinTestVersions
+import com.rickbusarow.kase.gradle.GradleTestEnvironment
+import com.rickbusarow.kase.gradle.KaseGradleTest
+import com.rickbusarow.kase.gradle.VersionMatrix
+import com.rickbusarow.kase.gradle.dsl.BuildFileSpec
+import com.rickbusarow.kase.stdlib.letIf
+import com.squareup.anvil.plugin.buildProperties.anvilVersion
+import com.squareup.anvil.plugin.buildProperties.fullTestRun
+import com.squareup.anvil.plugin.buildProperties.localBuildM2Dir
+import java.io.File
+
+abstract class BaseGradleTest(
+  override val versionMatrix: VersionMatrix = AnvilVersionMatrix(),
+) : KaseGradleTest<GradleKotlinTestVersions>,
+  FileStubs,
+  MoreAsserts {
+
+  override val localM2Path: File
+    get() = localBuildM2Dir
+
+  override val kases: List<GradleKotlinTestVersions>
```

Comment:
I get your point, but *within* the context of the library, the meaning is clear.  The property holds a `List<Kase>`, and in this case (😁) those kases are versions, but even in Gradle-specific tests that won't always be so.

I think the best I could do would be to introduce an alias property here.

Reply:
Hmm but the type is `GradleKotlinTestVersions`, that's why it feels like there's the disconnect between the name and the type. And I think an alias to have the type be `Kase` kinda goes in the opposite direction of what i was hoping because it introduces new terminology when there's already more "simple" terminology that can be used like the current type. IMO simpler, less context that is required to understand what is happening, the better. Not a hill I'm going to die on 😄, but I do feel like something like `versionCombos` better conveys what this is to any reader.

Reply:
What I mean is that this is the interface that defines the property:

```kotlin
public interface HasKases<out K : Kase> {
  public val kases: List<K>
}
```

So it doesn't just relate to versions.

Meanwhile, by `alias` (not `typealias` ;)) I mean adding a property to `BaseGradleTest` like this:

```kotlin
abstract class BaseGradleTest(
  override val versionMatrix: VersionMatrix = AnvilVersionMatrix(),
) : KaseGradleTest<GradleKotlinTestVersions>,
  FileStubs,
  MoreAsserts {

  override val kases: List<GradleKotlinTestVersions>
    get() = // ...

  /** an alias for [kases] */
  open val versions: List<GradleKotlinTestVersions>
    get() = kases
}
```